src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -1,186 +1,346 @@
+			CoreApplication.Resuming += CoreApplication_Resuming;
+		}
+
+		private void CoreApplication_Resuming(object sender, object e)
+		{
+			CoreDispatcher.Main.RunAsync(
+				priority: CoreDispatcherPriority.Normal,
+				handler: InitializeIfPermissionIsGranted
+			);
+		}
+
+		private async void InitializeIfPermissionIsGranted()
+		{
+			// If the user has granted the location permission while the app was in background, Initialize
+			if (await PermissionsHelper.CheckFineLocationPermission(CancellationToken.None))
+			{
+				TryInitialize();
+				CoreApplication.Resuming -= CoreApplication_Resuming;
+			}
+		}
+
+		private LocationManager InitializeLocationProvider(double desiredAccuracy)
+		{
+			var locationManager = (LocationManager)global::Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+
+			var criteriaForLocationService = new Criteria
+			{
+				Accuracy = Accuracy.Coarse
+			};
+
+			var acceptableLocationProviders = locationManager.GetProviders(criteriaForLocationService, true);
+
+			if (acceptableLocationProviders.Any())
+			{
+				_locationProvider = acceptableLocationProviders.First();
+			}
+			else
+			{
+				_locationProvider = String.Empty;
+			}
+
+			return locationManager;
+		}
+
+		partial void StartPositionChanged()
+		{
+			_positionChangedSubscriptions.TryAdd(this, 0);
+		}
+
+		partial void StopPositionChanged()
+		{
+			_positionChangedSubscriptions.TryRemove(this, out var _);
+		}
+
+		public async Task<Windows.Devices.Geolocation.Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
+		{
+
+			_locationManager = (LocationManager)Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+
+			_reportInterval = 1000;
+			_movementThreshold = 0;
+
+			RequestUpdates();
+
+			var providers = _locationManager.GetProviders(_locationCriteria, true);
+			int bestAccuracy = 10000;
+			Location bestLocation = null;
+
+			BroadcastStatus(PositionStatus.Initializing);
+
+			var startDate = DateTimeOffset.UtcNow;
+			foreach (string locationProvider in providers)
+			{
+				var location = _locationManager.GetLastKnownLocation(locationProvider);
+				if (location != null)
+				{
+					// check how old is this fix
+					var date = DateTimeOffset.FromUnixTimeMilliseconds(location.Time);
+					if (date + maximumAge > startDate)
+					{   // can be used, but is it best accuracy?
+						if (location.HasAccuracy)
+						{
+							if (location.Accuracy < bestAccuracy)
+							{
+								bestAccuracy = (int)location.Accuracy;
+								bestLocation = location;
+							}
+						}
+						else
+							bestLocation = location;
+					}
+				}
+			}
+
+			if (bestLocation != null)
+			{
+				RemoveUpdates();
+				BroadcastStatus(PositionStatus.Ready);
+				return bestLocation.ToGeoPosition();
+			}
+
+			// wait for fix
+			for (int i = (int)(timeout.TotalMilliseconds / 250.0); i > 0; i--)
+			{
+				await Task.Delay(250);

```

Comment:
Use a Stopwatch and while loop instead

Reply:
stopwatch?

Reply:
what do you mean by stopwatch?

Reply:
what do you mean by stopwatch?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch?view=net-5.0

Reply:
A while loop that uses the stopwatch.Elapsed and calls await `Task.Delay(250)` is fine.

Reply:
So, exactly same effect as my loop, but using more sophisticated tools. Ok.